sdk/src/azure/iot/az_iot_common.c

```diff
@@ -26,7 +26,7 @@ AZ_NODISCARD az_result az_iot_message_properties_init(
 {
   _az_PRECONDITION_NOT_NULL(properties);
   _az_PRECONDITION_VALID_SPAN(buffer, 0, true);
```

Comment:
Not a fan of using this one to do both checks, eh? :p

Reply:
haha no only because I like having separate, explicit purpose preconditions here. Makes it a little easier to read

Reply:
I agree.